### PR TITLE
net, admitter: Extract the mac-address validation for reuse

### PR DESCRIPTION
**Pull request by EdDev**
### What this PR does

The mac address validation is currently embedded in the VMI Spec admitter. In order to allow reuse of the mac validation for other specs, the validation logic is extracted out.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
```release-note
NONE
```

### CODE REVIEW

File: pkg/network/link/mac.go [hunk 1]
```diff
@@ -19,8 +19,29 @@
...
 const StaticMasqueradeBridgeMAC = "02:00:00:00:00:00"
 
 func IsReserved(mac string) bool {
 	return mac == StaticMasqueradeBridgeMAC
 }
+
+// ValidateMacAddress performs a validation of the address validity in terms of format and size.
+// An empty mac address input is ignored (i.e. is considered valid).
+func ValidateMacAddress(macAddress string) error {
+	if macAddress != "" {
```
Comment (orelmisan) [inline pkg/network/link/mac.go:36]: Please consider reversing the condition in order to return early and decrease indentation.

Comment (EdDev) [reply pkg/network/link/mac.go:36]: I think this is classified as a nit, but it does not worth debating on it any further.
Done.

**Comment by EdDev** [comment]
[change](https://github.com/kubevirt/kubevirt/compare/3054fe42e14a85c77dbed72b17ca1d64f746e33e..d0befc774a69d3a43c06e2d9474eb8a326ce63d6): Answered review comment.

**Review by orelmisan — APPROVED** [review]
Thank you

**Review by EdDev — COMMENTED** [review]
/approve

**Pull request merged**